Scripts/RemediationScripts/Remediate-ConfigureRoleBasedAPIAcessOnlyForAISearch.ps1

```diff
@@ -0,0 +1,1006 @@
+    Write-Host "Account Type: [$($context.Account.Type)]"
+    Write-Host $([Constants]::SingleDashLine)
+
+    # Note about the required access required for remediation
+
+    Write-Host "To reset API access for Azure AI Search services in a Subscription, Contributor or higher privileges on the Azure AI Search Services are required." -ForegroundColor $([Constants]::MessageType.Info)
+    Write-Host $([Constants]::SingleDashLine)
+    Write-Host "[Step 2 of 4] Preparing to fetch Azure AI Search services"
+    Write-Host $([Constants]::SingleDashLine)
+    if (-not (Test-Path -Path $FilePath))
+    {
+        Write-Host "Input file - [$($FilePath)] not found. Exiting..." -ForegroundColor $([Constants]::MessageType.Error)
+        Write-Host $([Constants]::DoubleDashLine)
+        return
+    }
+
+    Write-Host "Fetching all Azure AI Search services from [$($FilePath)]..." -ForegroundColor $([Constants]::MessageType.Info)
+    Write-Host $([Constants]::SingleDashLine)
+       
+    $AISearchServicesFromFile = Import-Csv -LiteralPath $FilePath
+    $validAISearchServices = $AISearchServicesFromFile | Where-Object { ![String]::IsNullOrWhiteSpace($_.Name) }
+    
+    $resource = @()
+    $resourceList = @()
+
+    $validAISearchServices | ForEach-Object {
+        $res = $_
+        $name = $_.Name
+        $resourceGroupName = $_.ResourceGroupName
+        $disableLocalAuthBeforeRemediation = $false
+        $disableLocalAuthAfterRemediation = $_.DisableLocalAuth
+        $authOption = $_.AuthOption
+
+        try
+        {
+            $resource = (Get-AzSearchService -Name $name  -ResourceGroupName $resourceGroupName -ErrorAction SilentlyContinue) 
+            $resourceList += $resource | Select-Object @{N='Name';E={$name}},
+                                                            @{N='ResourceGroupName';E={$resourceGroupName}},
+                                                            @{N='CurrentDisableLocalAuth';E={$true}},
+                                                            @{N='PreviousDisableLocalAuth';E={$false}}
+                                                                
+        }
+        catch
+        {
+            Write-Host "Error fetching Azure AI Search service : [$($serverName)]. Error: $($_)" -ForegroundColor $([Constants]::MessageType.Error)
+            Write-Host "Skipping this Azure AI Search service..." -ForegroundColor $([Constants]::MessageType.Warning)
+            Write-Host $([Constants]::SingleDashLine)
+        }
+    }
+        
+    # Includes Azure AI Search service
+    $AISearchservicesWithRBACOnlyAPIAccess= @()
+ 
+    Write-Host "[Step 3 of 4] Fetching Azure AI Search services"
+    Write-Host $([Constants]::SingleDashLine)
+    Write-Host "Separating Azure AI Search service where API access is RBAC only..." -ForegroundColor $([Constants]::MessageType.Info)
+    Write-Host $([Constants]::SingleDashLine)
+    $validAISearchServices | ForEach-Object {
```

Comment:
Is the $resourceList created in line 827 supposed to be used here instead of $validAISearchServices?
The resourceList isn't used anywhere.